### ISSUE TITLE
RFE: allow for optional kwargs fro input data

### ISSUE THREAD

**Issue by ankostis** [cedricleroy/pyungo#34]
- Python's kwargs are optional (since defaults are given in the function declaration).
- This library's *kwargs* feature is not default - it has to exist in the input-data or an error is raised.

These two facts cause a mismatch when converting traditional code into a graph-pipeline. 

If it is feasible, it would really help to add another `optional` keword in the `Graph.add_node()`.

## Steps to reproduce
The following code:
```python
graph = pyungo.Graph()

@graph.register(inputs=['a'], kwargs=['b'], outputs=['c'])
def f(a, b=2):
    return a + b

graph.calculate({'a': 1})
```
... raises ``PyungoError: The following inputs are needed: ['b']``
while the function is fully capable of working without `b`.

## Proposal
This should work:
```python
@graph.register(inputs=['a'], optional=['b'], outputs=['c'])
def f(a, b=2):
    return a + b

graph.calculate({'a': 1})
```
and produce `3`.

**Comment by cedricleroy** [comment]
Yes, there is a problem here, good catch. I think `kwargs` just need to be removed from the output checks, not sure we need an additional `optional` node argument here.

**Comment by ankostis** [comment]
:+1:

**Comment by veronicaguo** [comment]
Hey @cedricleroy, I did some prototyping and seem to have a working solution for this. :smile:

**Comment by cedricleroy** [comment]
Closed by #36 and #37, and available in [v0.8.1](https://github.com/cedricleroy/pyungo/releases/tag/v0.8.1). Thanks @veronicaguo!
